### PR TITLE
upgrade default TiDB version to v3.0.5 (cherry-pick #1132)

### PR DESCRIPTION
**Pull request by shonge**
### What problem does this PR solve? 
https://github.com/pingcap/tidb-operator/issues/1176
Cherry-pick update default tidb version from branch release -1.0
### Check List 

Tests 

 - Unit test
 - E2E test
 - Stability test
 - Manual test (add detailed scripts or steps below)
 - No code

Code changes

 - Has Helm charts change
 - Has Go code change
 - Has CI related scripts change
 - Has documents change

Side effects

Related changes

### Does this PR introduce a user-facing change?:
 
 ```release-note
upgrade default TiDB version to v3.0.5
 ```

### CODE REVIEW

File: tests/config.go [hunk 1]
```diff
@@ -90,8 +90,12 @@ func NewConfig() (*Config, error) {
 	flag.StringVar(&cfg.configFile, "config", "", "Config file")
 	flag.StringVar(&cfg.LogDir, "log-dir", "/logDir", "log directory")
 	flag.IntVar(&cfg.FaultTriggerPort, "fault-trigger-port", 23332, "the http port of fault trigger service")
+<<<<<<< HEAD
```
Comment (tennix) [inline tests/config.go]: It still has merge/rebase conflict.

File: tests/manifests/e2e/e2e.yaml [hunk 2]
```diff
@@ -7,8 +7,8 @@ metadata:
     app: webhook-service
 spec:
   ports:
-  - port: 443
-    targetPort: 443
+    - port: 443
```
Comment (tennix) [inline tests/manifests/e2e/e2e.yaml]: Could you remain the indentation as it is?

File: charts/tidb-cluster/values.yaml [hunk 3]
```diff
@@ -171,7 +171,8 @@ pd:
 
   # Specify the security context of PD Pod.
   # refer to https://kubernetes.io/docs/tasks/configure-pod-container/security-context/#set-the-security-context-for-a-pod
-  podSecurityContext: {}
+  podSecurityContext:
```
Comment (tennix) [inline charts/tidb-cluster/values.yaml]: Why change this?

**Comment by shonge** [comment]
So weird... I will check and cherry-pick again.

**Pull request merged**